rgdps/usecases/messages.py

```diff
@@ -0,0 +1,269 @@
+        ctx,
+        sender_user_id=sender_user_id,
+        page=page,
+        page_size=page_size,
+        include_deleted=include_deleted,
+    )
+
+    messages_resp = []
+    for message in messages:
+        user = await repositories.user.from_id(ctx, message.recipient_user_id)
+
+        if user is None:
+            continue
+
+        messages_resp.append(MessageResponse(message=message, user=user))
+
+    messages_count = await repositories.message.from_sender_user_id_count(
+        ctx,
+        sender_user_id=sender_user_id,
+        include_deleted=include_deleted,
+    )
+
+    return PaginatedMessagesResponse(
+        messages=messages_resp,
+        total=messages_count,
+    )
+
+
+async def create(
+    ctx: Context,
+    sender_user_id: int,
+    recipient_user_id: int,
+    subject: str,
+    content: str,
+) -> Message | ServiceError:
+
+    if recipient_user_id == sender_user_id:
+        return ServiceError.MESSAGES_INVALID_RECIPIENT
+
+    recipient = await repositories.user.from_id(ctx, recipient_user_id)
+    if recipient is None:
+        return ServiceError.MESSAGES_INVALID_RECIPIENT
+
+    friendship_check = await repositories.user_relationship.check_relationship_exists(
+        ctx,
+        user_id=sender_user_id,
+        target_user_id=recipient_user_id,
+        relationship_type=UserRelationshipType.FRIEND,
+    )
+
+    block_check = await repositories.user_relationship.check_relationship_exists(
+        ctx,
+        user_id=recipient_user_id,
+        target_user_id=sender_user_id,
+        relationship_type=UserRelationshipType.BLOCKED,
+    )
+
+    if block_check:
+        return ServiceError.MESSAGES_SENDER_BLOCKED
+
+    if recipient.message_privacy is UserPrivacySetting.PRIVATE:
+        return ServiceError.MESSAGES_MESSAGES_DISABLED
+
+    if recipient.message_privacy is UserPrivacySetting.FRIENDS and not friendship_check:
+        return ServiceError.MESSAGES_SENDER_NOT_FRIENDS
+
+    message = await repositories.message.create(
+        ctx,
+        sender_user_id=sender_user_id,
+        recipient_user_id=recipient_user_id,
+        subject=subject,
+        content=content,
+    )
+
+    return message
+
+
+async def mark_message_as_seen(
+    ctx: Context,
+    user: User,
+    message_id: int,
+) -> Message | ServiceError:
+    message = await repositories.message.from_id(
+        ctx,
+        message_id=message_id,
+        include_deleted=False,
+    )
+
+    if message is None:
+        return ServiceError.MESSAGES_NOT_FOUND
+
+    if message.recipient_user_id != user.id:
+        return ServiceError.MESSAGES_INVALID_RECIPIENT
+
+    message = await repositories.message.update_partial(
+        ctx,
+        message_id=message_id,
+        seen_ts=datetime.now(),
+    )
+
+    if message is None:
+        return ServiceError.MESSAGES_NOT_FOUND
+
+    return message
+
+
+async def delete_by_user(
+    ctx: Context,
+    user: User,
```

Comment:
I would probably just pass user_id here.

Reply:
resolved in 13c7f80